include/ibamr/PhaseChangeUtilities.h

```diff
@@ -0,0 +1,325 @@
+ *
+ * \param mu_idx a patch data index for the current viscosity variable maintained by the integrator.
+ * \param ctx is the pointer to SetFluidProperties class object.
+ */
+
+void callSetViscosityCallbackFunction(int mu_idx,
+                                      SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> > mu_var,
+                                      SAMRAI::tbox::Pointer<IBTK::HierarchyMathOps> hier_math_ops,
+                                      const int cycle_num,
+                                      const double time,
+                                      const double current_time,
+                                      const double new_time,
+                                      void* ctx);
+
+class SetFluidProperties
+{
+public:
+    /*!
+     * Constructor for this class.
+     */
+    SetFluidProperties(const std::string& object_name,
+                       SAMRAI::tbox::Pointer<IBAMR::AdvDiffHierarchyIntegrator> adv_diff_solver,
+                       const SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > H_var,
+                       SAMRAI::solv::RobinBcCoefStrategy<NDIM>* H_bc_coef,
+                       const SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > lf_var,
+                       SAMRAI::solv::RobinBcCoefStrategy<NDIM>* lf_bc_coef,
+                       const double rho_liquid,
+                       const double rho_solid,
+                       const double rho_gas,
+                       const double kappa_liquid,
+                       const double kappa_solid,
+                       const double kappa_gas,
+                       const double Cp_liquid,
+                       const double Cp_solid,
+                       const double Cp_gas,
+                       const double mu_liquid,
+                       const double mu_solid,
+                       const double mu_gas);
+
+    /*!
+     * Constructor for this class.
+     *
+     * This constructor can be used when only the energy equation is solved.
+     */
+    SetFluidProperties(const std::string& object_name,
+                       SAMRAI::tbox::Pointer<IBAMR::AdvDiffHierarchyIntegrator> adv_diff_solver,
+                       const SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > H_var,
+                       SAMRAI::solv::RobinBcCoefStrategy<NDIM>* H_bc_coef,
+                       const SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > lf_var,
+                       SAMRAI::solv::RobinBcCoefStrategy<NDIM>* lf_bc_coef,
+                       const double rho_liquid,
+                       const double rho_solid,
+                       const double rho_gas,
+                       const double kappa_liquid,
+                       const double kappa_solid,
+                       const double kappa_gas,
+                       const double Cp_liquid,
+                       const double Cp_solid,
+                       const double Cp_gas);
```

Comment:
As above --- don't `const` qualify in the interface when passing by value.

Reply:
(Also, I think that `const Pointer<T>` does not do what it seems like it should do.)

Reply:
I won't mark the other instances, but the same comment applies below as well...